src/tests/common.rs

```diff
@@ -140,12 +140,18 @@ async fn uni_directional_streams() -> Result<()> {
         assert_eq!(src, peer2_addr);
         src
     } else {
-        return Err(Error::UnexpectedMessageType);
+        panic!("Expected a unidirectional stream")
```

Comment:
Any reason we are going back to panic here instead of returning Result?

Reply:
We are not going back, but I think in this case `panic` is more appropriate because this is actually a form of assertion. It's just not expressed with the `assert` macro because there is no macro that asserts that something matches a pattern (there is `assert_matches` crate which we use in routing but I wasn't sure we want to add it here as well).

Reply:
>(there is assert_matches crate which we use in routing but I wasn't sure we want to add it here as well).
I would perhaps @madadam  if it allows us to get rid of a panic as assert is more obvious in tests IMO.

Reply:
OK, will use `assert_matches`.